src/kernel.zig

```diff
@@ -3,12 +3,31 @@ const uart = @import("uart.zig");
 
 // Here we set up a printf-like writer from the standard library by providing
 // a way to output via the UART.
-const Writer = std.io.Writer(u32, error{}, uart_put_str);
-const uart_writer = Writer { .context = 0 };
+pub fn drainUart(w: *std.io.Writer, data: []const []const u8, splat: usize) !usize {
+    _ = try uartPutStr(w.buffer);
+    w.end = 0;
+    var bytes_written: usize = 0;
+    for (data) |slice| {
+        bytes_written += try uartPutStr(slice);
+    }
+    if (splat == 0 or data.len < 1) return bytes_written;
+    const last_data = data[data.len - 1];
+    var i: usize = 0;
+    while (i < splat) : (i += 1) {
+        bytes_written += try uartPutStr(last_data);
+    }
+    return bytes_written;
```

Comment:
The last element of data must be written `splat` times which might be zero. This means that here's a subtle bug